### PR TITLE
[MOBILE-944] Update InboxMessageView

### PR DESCRIPTION
**Pull request by crow**
### What do these changes do?
Cleans up the InboxMessageView to be more like the react-native implementation. Implements this additional widget functionality in the sample.

### Why are these changes necessary?
These changes are necessary to improve the inbox message interface on the plugin for the first release.

### How did you verify these changes?
Manually tested.

#### Verification Screenshots:
https://urbanairship.slack.com/files/U0298VBRH/FPN4X9TBM/update_inboxmessageview.mov

### Anything else a reviewer should know?
This gets the timeboxed work in place, but there is some follow up work to do here related to how the plugin interfaces platform-specific web view loading:
https://urbanairship.atlassian.net/browse/MOBILE-987

### CODE REVIEW

File: lib/src/inbox_message_view.dart [hunk 1]
```diff
@@ -6,43 +6,51 @@ import 'package:flutter/widgets.dart';
 import 'airship.dart';
 
 class InboxMessageView extends StatefulWidget {
-  final InboxMessageViewCreatedCallback onViewCreated;
+  final InboxMessage message;
```
Comment (rlepinski) [inline lib/src/inbox_message_view.dart]: Require a message ID, not a message.

File: lib/src/inbox_message_view.dart [hunk 2]
```diff
@@ -6,43 +6,51 @@ import 'package:flutter/widgets.dart';
...
 
-  InboxMessageViewController.init(int id) {
+  Future<void> onPlatformViewCreated(id) async {
     _channel =  new MethodChannel('com.airship.flutter/InboxMessageView_$id');
+    if (widget.onLoadStarted != null) {
+      widget.onLoadStarted();
+    }
+    loadMessage(widget.message).then((message) {
+      if (widget.onLoadFinished != null) {
+        widget.onLoadFinished();
```
Comment (rlepinski) [inline lib/src/inbox_message_view.dart]: This is not really the load finish we are looking for. This should happen after the message is fully loaded into the web view.

File: lib/src/inbox_message_view.dart [hunk 3]
```diff
@@ -6,43 +6,51 @@ import 'package:flutter/widgets.dart';
...
+      }
+    });
   }
 
   Future<void> loadMessage(InboxMessage message) async {
     if (message == null) {
+      if (widget.onLoadError != null) {
+        widget.onLoadError(ArgumentError.notNull('message'));
```
Comment (rlepinski) [inline lib/src/inbox_message_view.dart]: Look at the react implementation. Should be from the native layer if it fails to load the message.

File: lib/src/inbox_message_view.dart [hunk 4]
```diff
@@ -63,4 +71,12 @@ class _InboxMessageViewState extends State<InboxMessageView> {
 
     return new Text('$defaultTargetPlatform is not yet supported by this plugin');
   }
+
+  dispose() {
+    super.dispose();
+
+    if (widget.onClose != null) {
+      widget.onClose();
```
Comment (rlepinski) [inline lib/src/inbox_message_view.dart]: This is not that the widget closed, but if `UAirship.close()` is called from within the message view.

File: lib/src/inbox_message_view.dart (hunk 1 above)
Comment (crow) [reply lib/src/inbox_message_view.dart]: added this

File: lib/src/inbox_message_view.dart [hunk 5]
```diff
@@ -3,50 +3,21 @@ import 'dart:async';
...
-
-class _InboxMessageViewState extends State<InboxMessageView> {
-
   Future<void> onPlatformViewCreated(id) async {
-    if (widget.onViewCreated == null) {
-      return;
-    }
-    widget.onViewCreated(new InboxMessageViewController.init(id));
+    MethodChannel _channel =  new MethodChannel('com.airship.flutter/InboxMessageView_$id');
+    _channel.invokeMethod('loadMessage', messageId);
```
Comment (rlepinski) [inline lib/src/inbox_message_view.dart]: where is it handling the message ID changing?

Comment (crow) [reply lib/src/inbox_message_view.dart]: the inbox_message_view is a stateless and will build and reload the message when the messageId changes

File: example/lib/screens/message_view.dart [hunk 6]
```diff
@@ -2,33 +2,60 @@ import 'package:flutter/material.dart';
...
+            list = List<InboxMessage>.from(snapshot.data);
+          }
+
+          message = list.firstWhere((thisMessage) =>
+          widget.messageId == thisMessage.messageId,
+              orElse: () => null);
+
+          return Scaffold(
+            appBar: AppBar(
+              title: message != null ? Text("${message.title}") : Container(),
```
Comment (BrianBatchelder) [inline example/lib/screens/message_view.dart]: Is the view empty if the message is nil? Is there an alert like on iOS? Does it just show the inbox?

Comment (crow) [reply example/lib/screens/message_view.dart]: The view body is empty until the web page request completes. The message property here is null until the message list future returns. In this case, we just show an invisible container until the title is ready to be displayed.

**Pull request merged**